doc/cephfs/posix.rst

```diff
@@ -0,0 +1,37 @@
+========================
+ Differences from POSIX
+========================
+
+CephFS aims to adhere to POSIX semantics wherever possible.  For
+example, in contrast to many other common network file systems like
+NFS, CephFS maintains strong cache coherency across clients.  The goal
+is for processes communicating via the file system to behave the same
+when they are on different hosts as when they are on the same host.
+
+However, there are a few places where CephFS diverges from strict
+POSIX semantics for various reasons:
+
+- In shared simultaneous writer situations, a write that crosses
```

Comment:
If you have read+write or write+write we flip to LOCK_MIX mode, which gives out RD and WR caps but makes all IO synchronous (on RDCACHE or WRBUFFER).  All reads and writes are effectively serialized at the OSD.

CEPH_IOC_LAZY asks the mds to give the client RDCACHE and WRBUFFER despite the sharing, which means the app is responsible for invalidating its cache or flushing changes at the appropriate points...